include/openthread/thread.h

```diff
@@ -811,9 +811,12 @@ void otThreadRegisterParentResponseCallback(otInstance *                   aInst
  */
 typedef struct otThreadDiscoveryRequestInfo
 {
-    otExtAddress mExtAddress;   ///< IEEE 802.15.4 Extended Address of the requester
-    uint8_t      mVersion : 4;  ///< Thread version.
-    bool         mIsJoiner : 1; ///< Whether is from joiner.
+    otExtAddress mExtAddress;                            ///< IEEE 802.15.4 Extended Address of the requester
+    uint8_t      mVersion : 4;                           ///< Thread version.
+    bool         mIsJoiner : 1;                          ///< Whether is from joiner.
+    uint32_t     mOui;                                   ///< The vendor Oui
```

Comment:
I am fine with the PR and this approach (for first implementation)

Just want to bring up a topic related to this (to consider for future)

We may have more than one Joiner Adv info (for different OUIs) in the same discovery message. How to handle this situation?
- One way is to make this a linked-list or array of "Joiner Adv Info".
- Second way is to invoke the callback multiple times (for same received discovery msg) with different `otThreadDiscoveryRequestInfo` entries (each time for a different OUI) - I would prefer this.